### PR TITLE
Refactor utils according to new prelude

### PR DESCRIPTION
**Pull request by NianhengWu**
Since we've already made prelude great again, there should be some corresponding changes in utils too.

### CODE REVIEW

File: src/convert.rs [hunk 1]
```diff
@@ -118,7 +122,7 @@ impl FinalfusionApp for ConvertApp {
             .metadata_filename
             .as_ref()
             .map(read_metadata)
-            .map(Metadata);
+            .map(|metadata| Metadata::new(metadata));
```
Comment (danieldk) [inline src/convert.rs]:
```rust
.map(Metadata::new);
```

should be enough

File: src/metadata.rs [hunk 2]
```diff
@@ -48,7 +50,7 @@ impl FinalfusionApp for MetadataApp {
         if let Some(metadata) = read_metadata(&self.input_filename) {
             writer
                 .write_all(
-                    to_string_pretty(&metadata.0)
+                    to_string_pretty(&metadata.deref())
```
Comment (danieldk) [inline src/metadata.rs]: did you try

```
to_string_pretty(&metadata)
```

? In some cases deref coercion kicks in.

Comment (NianhengWu) [reply src/metadata.rs]: Yeah I tried, but ```serde::ser::Serialize``` is not implemented for ```Metadata```, so it doesn't work :(

Comment (danieldk) [reply src/metadata.rs]: Ok, then you could use:

```
to_string_pretty(&*metadata)
```

**Pull request merged**